src/run/headers.ts

```diff
@@ -272,7 +272,10 @@ export const setCacheControlHeaders = (
 }
 
 export const setCacheTagsHeaders = (headers: Headers, requestContext: RequestContext) => {
-  if (requestContext.responseCacheTags) {
+  if (
+    requestContext.responseCacheTags &&
+    (headers.has('cache-control') || headers.has('netlify-cdn-cache-control'))
```

Comment:
this is to prevent setting tags on uncacheable responses when fallback in next@canary is served